src/DrDotnet.Profilers/report.rs

```diff
@@ -1,71 +1,53 @@
-use serde::{Deserialize, Serialize};
-use uuid::Uuid;
-use chrono::{DateTime, Local};
-
 use std::path::{PathBuf, Path};
 use std::io::BufWriter;
 use std::fs::File;
 use std::io::Write;
 
-use crate::profilers::ProfilerData;
+use crate::rust_protobuf_protos::interop::*;
 
-// A Session refers to a profiling session. In conists in a process, a profiler, and a timestamp at which the profiling was done.
-#[derive(Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct Session {
-    session_id: Uuid,
-    process_name: String,
-    timestamp: DateTime<Local>,
-    profiler: ProfilerData,
+pub trait Session {
+    fn create_session_json(&self);
+    fn create_report(&self, filename: String) -> Report;
+    fn get_root_directory() -> String;
+    fn get_directory(session_id: &String) -> String;
 }
 
-impl Session {
+impl Session for SessionInfo {
 
     // Returns a Session from its UID and ProfilerData.
     // If the Session report is not present on the disk, it will be written at the same time.
-    pub fn get_session(session_id: Uuid, profiler: ProfilerData) -> Session {
-
-        let process_name = std::env::current_exe().unwrap()
-            .file_name().unwrap()
-            .to_str().unwrap()
-            .to_owned();
+    fn create_session_json(&self) {
 
-        let report = Session {
-            session_id: session_id,
-            process_name : process_name,
-            profiler: profiler,
-            timestamp: chrono::offset::Local::now()
-        };
+        let s = SessionInfo::new();
 
         // Serialize to JSON
-        let json = serde_json::to_string_pretty(&report).unwrap();
+        let json = protobuf_json_mapping::print_to_string(&s).unwrap();
```

Comment:
Err I might be missing something but why are we creating a new `SessionInfo` with  `let s = SessionInfo::new();` instead of passing `&self` to:
`let json = protobuf_json_mapping::print_to_string(&self).unwrap();`

Reply:
It was a test to see if you were following 🥸😅

Reply:
:fixed: